lib/viewmodels/user_centauri_points_view_model.dart

```diff
@@ -0,0 +1,41 @@
+import "dart:async";
+
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/models/database/user/user_id_firestore.dart";
+import "package:proxima/services/database/user_repository_service.dart";
+
+/// A view model that provides the number of centauri points of a user
+/// given its user id. Allow a null [arg] (user id) to return null, which
+/// is useful for loading states.
+class UserCentauriPointsViewModel
+    extends FamilyAsyncNotifier<int?, UserIdFirestore?> {
+  UserCentauriPointsViewModel();
+
+  @override
+  Future<int?> build(UserIdFirestore? arg) async {
+    final userDataBase = ref.watch(userRepositoryServiceProvider);
+
+    if (arg == null) {
+      return null;
+    }
```

Comment:
You are completely right, however I think it relies on doing a conditional `watch` in the UI with the current state of the code, which is not desired. I think that doing a clean solution is much wider than this PR's scope and should be addressed in a separate PR, along with the potential merge of `DynamicUserAvatar` and `UserAvatar` and the removal of nullable user IDs. 👍🏼 

Tell me what you think: 💯

Reply:
Mhm I see indeed. This would require a big refactor to never pass a null id; which is not a choice you took in this PR. This makes sense ok, you can stay like this.

Reply:
I agree. 

Modified in: [0e38ef4](https://github.com/ProximaEPFL/proxima/pull/348/commits/0e38ef4bcd244327ec37406349edaacc520318b2)

Thank you ! 

Note that I made many changes, let us continue this in the main thread as I am going to ask @yoannLafore to re-review as a big chunk changed 👍🏼

Reply:
Note that I added the task of accepting a non-nullable `userId` in the dynamic user avatar view model: [#353](https://github.com/ProximaEPFL/proxima/issues/353)

Reply:
You refactor is perfect, great job! :)

Yes, I agree with the fact that creating a new task to the backlog for the `userId` is the good choice, you've already done a lot in this PR; and this was not a choice you took in this PR. Great job :)